lib/Sema/CSApply.cpp

```diff
@@ -7295,39 +7302,104 @@ static Expr *finishApplyDynamicCallable(ConstraintSystem &cs,
   bool useKwargsMethod = argumentLabel == ctx.Id_withKeywordArguments;
 
   // Construct expression referencing the `dynamicallyCall` method.
-  Expr *member =
+  MemberRefExpr *member =
     new (ctx) MemberRefExpr(fn, fn->getEndLoc(), ConcreteDeclRef(method),
                             DeclNameLoc(method->getNameLoc()),
                             /*Implicit*/ true);
 
+  cs.setType(member, methodType);
+
+  bool isDynamic
+      = selected.choice.getKind() == OverloadChoiceKind::DeclViaDynamic;
+  Expr* memberExpr = buildMemberRef(
+      member->getBase(), selected.openedFullType, member->getDotLoc(),
+      selected.choice, member->getNameLoc(), selected.openedType,
+      cs.getConstraintLocator(member), loc, true,
+      selected.choice.getFunctionRefKind(), member->getAccessSemantics(),
+      isDynamic);
+
   // Construct argument to the method (either an array or dictionary
   // expression).
   Expr *argument = nullptr;
   if (!useKwargsMethod) {
-    argument = ArrayExpr::create(ctx, SourceLoc(), arg->getElements(),
+    auto paramArrayType = methodType->getParams()[0].getParameterType();
+    auto paramType = dyn_cast<ArraySliceType>(paramArrayType.getPointer())->getBaseType();
```

Comment:
I would use the desugared form instead, in case the type becomes `Array<T>` and not `[T]`.

Reply:
Whoops. This should use the conformance anyways.